app/src/main/java/ar/com/wolox/android/cookbook/tests/TestLoginRecipePresenter.kt

```diff
@@ -6,6 +6,34 @@ import javax.inject.Inject
 class TestLoginRecipePresenter @Inject constructor(private val loginService: TestLoginRecipeService) : BasePresenter<TestLoginRecipeView>() {
 
     fun onLoginButtonClick(email: String, password: String) {
-        loginService.login(email, password)?.let { view.goToNextWindow() } ?: view.showLoginError()
+        val isValidEmail = verifyEmail(email)
+        val isValidPassword = verifyPassword(password)
+
+        if (isValidEmail && isValidPassword) {
+            loginService.login(email, password, { view.goToNextWindow() }, { view.showLoginError() })
+        }
+    }
+
+    private fun verifyEmail(email: String): Boolean {
```

Comment:
This function should only return whether the `email` is valid and the caller show whatever it has to show. It's doing too much.